File: solutions/0347-top-k-frequent-elements/solution.py
class Solution:
    def topKFrequent(self, nums: List[int], k: int) -> List[int]:
        map = {}
        freq = [[] for i in range(len(nums)+1)]

        for n in nums: 
            map[n] = map.get(n, 0) + 1
        for n, c in map.items():
            freq[c].append(n)       

        result = []

        for i in range (len(freq) - 1, 0, -1):
            for j in range (len(freq[i])):
                result.append(freq[i][j])
                if (len(result) == k):
                    return result     

       

File: solutions/0347-top-k-frequent-elements/solution.js
/**
 * @param {number[]} nums
 * @param {number} k
 * @return {number[]}
 */
var topKFrequent = function(nums, k) {
    let result = []
    let map = new Map()

    for (n of nums) {
        map.set(n, (map.get(n) || 0) + 1)
    }

    const sortedMap = new Map([...map.entries()].sort((a, b) => b[1] - a[1]))

    for (item of sortedMap.keys()) {
        if (result.length < k) {
            result.push(item)
        }
    }

    return result
};
